File: Chapter 03/JS/section_3_3_1_srp/after/player.js
/**
 * Responsibility 1: State and Core Actions.
 * * ARCHITECTURE NOTE: By stripping out the database and tactical logic, the Player 
 * class is now highly cohesive. It has only one reason to change: if the fundamental 
 * definition of a player (like adding a 'pass' method or 'stamina' property) changes.
 * * This is now a "POJO" (Plain Old JavaScript Object) style entity that is 
 * incredibly easy to move between different parts of the system.
 */
class Player {
    constructor(name) {
        this.name = name;
    }

    /**
     * Executes a core domain action.
     */
    dribbleBall() {
        console.log(`  [Action] ${this.name} is dribbling the ball down the court.`);
    }
}

module.exports = Player;

File: Chapter 03/C#/Section_3_3_1_SRP/After/Player.cs
using System;

namespace Chapter03.SRP.After;

/// <summary>
/// Responsibility 1: State and Core Actions.
/// 
/// ARCHITECTURE NOTE: By stripping out the database and tactical logic, the Player 
/// class is now highly cohesive. It has only one reason to change: if the fundamental 
/// rules of a player (like adding a 'Pass' method or 'Stamina' property) change.
/// </summary>
public class Player
{
    public required string Name { get; set; }
    
    /// <summary>
    /// Executes a core domain action specific to the player's physical state.
    /// </summary>
    public void DribbleBall() 
    {
        Console.WriteLine($"  [Action] {Name} is dribbling the ball down the court.");
    }
}